### PR TITLE
Add fixtures and sample functional tests

### PR DESCRIPTION
**Pull request by lizadaly**
The goal was to add enough fixtures to the functional tests to enable useful browser-level testing (and then to apply that to the printable HTML view). 

This adds just enough tests to ensure that a user can log in, interact with Vue, and explore a Casebook.

## Long note follows

This is one of those things that looks more straightforward than it was.

It turned out that the `live_server` pytest fixture had never been serving up static assets, though that was not apparent in the original functional test. The static asset URLs are in the production form of `/static/dist/main-$hash.css` and that exact asset did live on the filesystem, but the test server refused to complete the request so no styles or javascript were loading.

A Github comment led me to the solution, which is that our particular use of bundled assets _and_ `staticfiles` led the `live_server` pytest fixture to operate in the wrong mode. The original solution and proposed upstream PR allow the fixture to delay its construction until a user can decide which static asset method to use. I found a different solution, which was to use a new fixture that drops the static files app before the `live_server` fixture is instantiated, then returns the fixture now operating under the correct assumptions. (There's some [extra trickiness](https://pytest-django.readthedocs.io/en/latest/helpers.html#live-server) around ensuring that both the fixture loading and live server database are actually the same database.)

Another option would be to drop `staticfiles` in `settings_pytest.py`, but then that would globally apply to all tests and be pretty hard to debug if it had some weird side effect. My preference here is to keep the override close to the code it's modifying. Hopefully at some point we can use the upstream fix method when that PR is eventually merged.

### CODE REVIEW

File: web/requirements.in [hunk 1]
```diff
@@ -25,7 +25,7 @@ django-simple-history # history preservation for contents
 
 # Testing
 pytest
-pytest-django
+pytest-django>=4.5
```
Comment (lizadaly) [inline web/requirements.in]: Both sets of dependencies here were too old to support the functionality needed here, so these got bumped.

**Comment by rebeccacremona** [comment]
The `staticfiles` thing is such a pain! Ran into it on capstone too, and ended up doing the [`settings_pytest.py` thing](https://github.com/harvard-lil/capstone/pull/2090/files#diff-0d63776784b6bcb413f653ec885342d1387b1f0b170ff857a6173d45b5666332R27) despite the risk of side effects. I knew then, but don't remember now, why Perma is different. Making a mental node (maybe an actual note if I can decide where it belongs) that we should probably use the same approach everywhere at some point... though maybe the upstream fix will come first 😀

**Comment by lizadaly** [comment]
Confirmed locally that the fixtures seem to be barfing up the non-playwright tests, so moving back to draft to fix that up.

**Comment by lizadaly** [comment]
>  I knew then, but don't remember now, why Perma is different.

I should've asked! I did look at Perma, but seeing nothing in its configuration that looked relevant I just assumed that the particular combination of staticfiles/whitenoise/django-webpack was different here and so nobody had hit the problem yet.

**Comment by lizadaly** [comment]
Tragically, it's the version bump in `pytest-django` that broke the existing doctest tests, rather than anything to do with this PR.

**Comment by lizadaly** [comment]
These tests are now fully passing after merging in #1803 so this is ready for review on its own merits.

**Review by kilbergr — APPROVED** [review]
LGTM!
I do get a failing test in `test_export`, unrelated to these changes:

```
../../usr/local/lib/python3.9/site-packages/django/apps/registry.py:91
  /usr/local/lib/python3.9/site-packages/django/apps/registry.py:91: RemovedInDjango41Warning: 'webpack_loader' defines default_app_config = 'webpack_loader.apps.WebpackLoaderConfig'. Django now detects this configuration automatically. You can remove default_app_config.
    app_config = AppConfig.create(entry)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
==================================================== short test summary info =====================================================
FAILED main/test/test_export.py::test_export - AssertionError: e1.text != e2.text:
```

**Comment by lizadaly** [comment]
Thanks for finding that error—this is another case where the existing test relied on the primary key value of some test data. 

The fixture tests are transactional and are rolled back at the end of their runs, but rollbacks do not reset the primary key sequence in postgres, they just delete the data represented by those rows.

When the tests are run parallelized (as in CI) the export test happened to come first and had the expected PK values. You probably just ran `pytest`, which ran each test in series. That happens to put the new fixture tests first, and by the time the export test ran, its new rows were created with slightly higher primary key values than expected. I was able to verify this locally by running just `pytest` myself.

The workaround here is add an existing fixture to reset the primary key sequences before test start, so now they start from zero.

**Pull request merged**